lib/backend/k8s/resources/profile.go

```diff
@@ -222,9 +224,40 @@ func (c *profileClient) EnsureInitialized() error {
 }
 
 func (c *profileClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
-	resl := list.(model.ResourceListOptions)
-	if len(resl.Name) != 0 {
-		return nil, fmt.Errorf("cannot watch specific resource instance: %s", list.(model.ResourceListOptions).Name)
+	// Build watch options to pass to k8s.
+	opts := metav1.ListOptions{Watch: true}
+	rlo, ok := list.(model.ResourceListOptions)
+	if !ok {
+		return nil, fmt.Errorf("ListInterface is not a ResourceListOptions: %s", list)
+	}
+
+	// Setting to Watch all profiles in all namespaces; overriden below
+	namePrefix := ""
+	ns := kapiv1.NamespaceAll
+	sa := ""
+
+	// Watch a specific profile.
+	if len(rlo.Name) != 0 {
+		log.WithField("name", rlo.Name).Debug("Watching a single profile")
+		if strings.HasPrefix(rlo.Name, conversion.NamespaceProfileNamePrefix) {
+			namePrefix = conversion.NamespaceProfileNamePrefix
+			var err error
+			ns, err = c.ProfileNameToNamespace(rlo.Name)
+			if err != nil {
+				return nil, err
+			}
+			opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", ns).String()
+		} else if strings.HasPrefix(rlo.Name, conversion.ServiceAccountProfileNamePrefix) {
+			namePrefix = conversion.ServiceAccountProfileNamePrefix
+			var err error
```

Comment:
nit: Let's just pre-declare 'err' outside of this block (up where we declare everything else) so we don't need to do it twice down here.